frontend/src/concepts/pipelines/content/pipelinesDetails/pipelineRun/runLogs/LogsTab.tsx

```diff
@@ -134,31 +167,72 @@ const LogsTabForPodName: React.FC<{ podName: string }> = ({ podName }) => {
                 'Error loading logs'
               ) : !logsLoaded ? (
                 <>
-                  <Spinner size="sm" /> Loading log
+                  <Spinner size="sm" /> {windowWidth > 576 ? 'Loading log' : 'Loading'}
                 </>
               ) : isPaused ? (
                 <>
-                  <PlayIcon /> Resume refreshing
+                  <PlayIcon /> {windowWidth > 576 ? 'Resume refreshing' : 'Resume'}
                 </>
               ) : (
                 <>
-                  <PauseIcon /> Pause refreshing
+                  <PauseIcon /> {windowWidth > 576 ? 'Pause refreshing' : 'Pause'}
                 </>
               )}
             </Button>
           </SplitItem>
           <SplitItem isFilled style={{ textAlign: 'right' }}>
-            <Tooltip position="top" content={<div>Download currently selected container</div>}>
-              <Button
-                onClick={onDownload}
-                variant="link"
-                aria-label="Download current logs"
-                icon={<DownloadIcon />}
-                isDisabled={!canDownload}
-              >
-                Download
-              </Button>
-            </Tooltip>
+            {downloading ? (
+              <>
+                <Spinner size="sm" />{' '}
+              </>
+            ) : (
+              ''
+            )}
+            {podContainers.length !== 0 ? (
+              <Dropdown
+                removeFindDomNode
+                toggle={
+                  windowWidth > 576 ? (
+                    <DropdownToggle
+                      id="download-steps-logs-toggle"
+                      onToggle={() => setIsDownloadDropdownOpen(!isDownloadDropdownOpen)}
+                    >
+                      Download
+                    </DropdownToggle>
+                  ) : (
+                    <KebabToggle
+                      onToggle={() => setIsDownloadDropdownOpen(!isDownloadDropdownOpen)}
+                    />
+                  )
+                }
+                isOpen={isDownloadDropdownOpen}
+                isPlain={windowWidth < 576}
```

Comment:
Don't write the same number multiple times. "Magic numbers" as they are called are the bane of maintaining code. 

If your check is "is this small screen" -- make a utility that checks.

```
const isSmallScreen = () => windowWidth < 576;

// examples
isPlain={isSmallScreen()}
content={isSmallScreen() ? 'small screen content' : 'larger screen content'}
<PauseIcon /> {isSmallScreen() ? 'Pause' : 'Pause refreshing'}
```
We don't have a lot of responsive design manually done like this... so we can just go with this for now.